File: SDKs/amazonaws.com/outposts/2019-12-03/java/lib/src/main/java/openapisdk/models/shared/LineItem.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * LineItem
 * Information about a line item.
**/
public class LineItem {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("CatalogItemId")
    public String catalogItemId;
    public LineItem withCatalogItemId(String catalogItemId) {
        this.catalogItemId = catalogItemId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("LineItemId")
    public String lineItemId;
    public LineItem withLineItemId(String lineItemId) {
        this.lineItemId = lineItemId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Quantity")
    public Long quantity;
    public LineItem withQuantity(Long quantity) {
        this.quantity = quantity;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Status")
    public String status;
    public LineItem withStatus(String status) {
        this.status = status;
        return this;
    }
}

File: SDKs/amazonaws.com/outposts/2019-12-03/python/src/sdk/models/shared/lineitem.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class LineItem:
    r"""LineItem
    Information about a line item.
    """
    
    catalog_item_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('CatalogItemId') }})
    line_item_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('LineItemId') }})
    quantity: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Quantity') }})
    status: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Status') }})
    
